core/src/main/java/org/apache/calcite/util/RelToSqlConverterUtil.java

```diff
@@ -93,6 +101,41 @@ public static void unparseSparkArrayAndMap(SqlWriter writer,
     writer.endList(frame);
   }
 
+
+  /**
+   * Rewrite SINGLE_VALUE use SparkSQL Mode.
+   *
+   * <blockquote><pre>
+   * CASE COUNT(*)
+   * WHEN 0 THEN NULL
+   * WHEN 1 THEN MIN(&lt;result&gt;)
+   * ELSE (SELECT NULL UNION ALL SELECT NULL))
```

Comment:
1. A problem: `SELECT NULL UNION ALL SELECT NULL` will return two NULL values ​​instead of one.
2. Suggestion: Use `COUNT(field)` instead of `COUNT(*)`

Reply:
> 1. SELECT NULL UNION ALL SELECT NULL

this is intentional to make sure when the subquery include more then one value ,This SQL will throw exception.

2: Suggestion: Use COUNT(field) instead of COUNT(*)

Please check CALCITE-6430.

Reply:
You should add a comment in the code about the UNION ALL. Your intent is not obvious at all.

I am not sure how the runtime error will look; if the error will mention a UNION ALL operator, the users may be confused, since there is no UNION ALL in the original query, but I guess there isn't much you can do about this.

Reply:
> > 1. SELECT NULL UNION ALL SELECT NULL
> 
> this is intentional to make sure when the subquery include more then one value ,This SQL will throw exception.
> 
> 2: Suggestion: Use COUNT(field) instead of COUNT(*)
> 
> Please check [CALCITE-6430](https://issues.apache.org/jira/browse/CALCITE-6430).

got it, thanks. Based on count(*) and union all, the result of the subquery is guaranteed to be only one row.

Reply:
Please add the explanations you gave here as comments in the code.
It will make maintaining the code much easier.